### DISCUSSION TITLE
Why livewire V3 always disable ConvertEmptyStringsToNull and TrimStrings middleware ? it is important validate in laravel

### DISCUSSION THREAD

**Discussion by firer777** [livewire/livewire#8357] [1. Help]
<img width="937" alt="Screen Shot 2024-04-27 at 00 37 26" src="https://github.com/livewire/livewire/assets/92707054/594ac320-1271-4d9e-ab2e-4ae2e21e0303">

**Comment by rafaelindriago** [comment]
I think is because the ```wire:model.live``` functionallity will have unexpected behaviour if you type something like "My ", that blank space will be removed and then you'll be unable to type something with blank spaces in "live" mode.

My personal solution is to implement a trait with hooks:
```
<?php

namespace App\Livewire\Traits;

use Illuminate\Support\Str;

trait WithTrimStrings
{
    public function updatedWithTrimStrings(string $property, mixed $value): void
    {
        if (is_string($value)) {
            $this->fill([
                $property => Str::trim($value),
            ]);
        }
    }
}
```

You can add an ```array``` with exceptions, like the ```TrimStrings``` middleware, to avoid the triming on "live" properties.

**Reply by firer777** [reply]
i did write helper function for trim and convert empty string to null input after validation and send data to save datatabse, no need change every thing of property
Example php helpers function: cleanArray(array $input) {.......}
Livewire:
public function store() 
{
     $this->validate();
     Product::create(cleanArray($this->data)); 
}
